hooks/post_gen_project.py

```diff
@@ -1,9 +1,211 @@
+import os
+import json
+import shutil
 import subprocess
+from os.path import join
 
-if __name__ == "__main__":
-{% if cookiecutter.create_repo == "yes" %}
-    subprocess.check_call(['git', 'init'])
-    subprocess.check_call(['hub', 'create'])
-{%- else %}
+import git
+
+from python_terraform import Terraform
+
+try:
+    input = raw_input  # py2 compatibility
+except NameError:
     pass
-{% endif %}
+
+TF_MODULE_PATH = (
+    "git::https://github.com/plus3it/terraform-github-repo.git?ref=0.0.0")
+YES_OPTIONS, NO_OPTIONS = frozenset(["y", "yes"]), frozenset(["n", "no", ""])
+PERM_OPTIONS = frozenset(["push", "pull", "admin"])
+
+
+def multiple_choices(question):
+    selection = []
+    additional = "Would you like to add another (y/n)? [n]: "
+
+    check = True
+    while check:
+        choice = input(question).lower()
+        selection.append(choice)
+        check = check_choice(additional)
+    return selection
+
+
+def check_choice(question):
+    while True:
+        choice = input(question).lower()
+        if choice in YES_OPTIONS:
+            return True
+        if choice in NO_OPTIONS:
+            return False
+
+
+def get_collaborators():
+    # Questions that have multiple layers to them
+    collaborators = []
+    collab_q = "Grant an organizational team access to the repo(y/n)? [n]: "
+
+    if check_choice(collab_q):
+        add_team = True
+
+        while add_team:
+            teams_q = "What team would you like to grant access? "
+            team = input(teams_q).lower()
+
+            if team != "":
+                perm = None
+                while perm not in PERM_OPTIONS:
+                    perms_q = (
+                        "What permissions would you like to grant them"
+                        + "(push, pull, admin)? ")
+                    perm = input(perms_q).lower()
+                collaborators.append({"name": team, "permission": perm})
+
+            another_q = "Would you like to add another team (y/n)? [n]: "
+            add_team = check_choice(another_q)
+
+    return collaborators
+
+
+def run_terraform(directory, target_module):
+    terraform = Terraform(directory)
+    terraform.init(from_module=target_module)
+    ret_code, _, _ = terraform.apply(auto_approve=True, capture_output=False)
+
+    return ret_code
+
+
+def open_pr(source_repo, working_dir):
+    """Opens a pull request."""
+
+    # clone the repo
+    git.Repo.clone_from(
+        source_repo,
+        "tempdir"
+        )
+
+    # Move the cloned contents up a directory
+    # because we can't clone to an existing directory
```

Comment:
We have `working_dir`... any reason not to just delete it before the clone, then clone to `working_dir`?

Reply:
working_dir = the directory that has been populated with the cookiecutter filled template. This is really just a cheater way of getting git configured properly by copying the contents of `.git`.

An alternative to moving files around/using a temp folder could be git initializing the working dir and just setting the new module's git repo as a remote. Kinda prefer what's here currently given that it's working but if the alternative is preferred I can implement it.

Reply:
Ok, I see. We've run terraform to create the GitHub repo, but then need to sync the local templated dir with the remote. If we stick with the tempdir thing, then let's use python's actual temp functionality...

https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp

```
temp_dir = tempfile.mkdtemp()
```

Alternative workflow ought to just look like this though, I think? Seems reasonably clean?

```
git init
git remote add origin <remote>
git fetch
git branch init origin/master  # this is key, the new branch should track origin/master
git checkout init
git add -all
...
```